### PR TITLE
[55] hot reload fix + console error fix

### PR DESCRIPTION
**Pull request by liuhx1027**
- add auto-generated files to .gitignore
- add key attribute to list element to remove console error
- support redux dev tool
- code auto re-formating

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,10 @@ npm-debug.log
 secrets.js
 /webpack.config.js
 /dist/bundle.js
+
+log.txt
+pouch__all_dbs__
+crops
+_users
+_replicator
+/docs/generated/*
```
Comment (petteripitkanen) [inline .gitignore]: `docs/generated/` are meant to be committed as they are exported to https://ecohackerfarm.github.io/powerplant/, so this line should be removed?

File: client/index.js [hunk 2]
```diff
@@ -37,3 +37,8 @@ class AppProvider extends React.Component {
 //addAllStyles();
 
 render(<AppProvider />, document.getElementById('app'));
+if (process.env.NODE_ENV === 'development' && module.hot) {
```
Comment (petteripitkanen) [inline client/index.js]: Checking `module.hot` is enough here I think. When HMR is not wanted then `HotModuleReplacementPlugin` shouldn't be included by webpack. `process.env.NODE_ENV` is more for the server side.

**Comment by petteripitkanen** [comment]
Thanks! It's working ok, would you still check these small issues that I found?

**Comment by liuhx1027** [comment]
@petteripitkanen , all issues from your comments had been fix.

**Pull request merged**